workers/task-driver/src/helpers.rs

```diff
@@ -76,12 +76,22 @@ pub fn transition_order_settling(order_id: OrderIdentifier, state: &State) -> Re
 }
 
 /// Update an order's state to `Filled`
-pub fn transition_order_filled(order_id: OrderIdentifier, state: &State) -> Result<(), String> {
+pub fn transition_order_filled(
```

Comment:
nit: maybe we should rename this / fix the doc comment, it implies that we're definitely marking the order as filled. perhaps `record_order_fill` or something?